### PR TITLE
Added MouseEvents AND changed window resize

### LINKED ISSUE #121: Windows as target platform
As games who want to be released through [Steam Greenlight](https://steamcommunity.com/greenlight/faq/), Windows support is kinda a must. 

However, `engi` currently doesn't build with `GOOS=windows`.

### PR DESCRIPTION
**Pull request by EtienneBruines**
Also added a demo for the mouse events. See #59.

The function `engi.SetScaleOnResize` allows user to say:
"I want things to become bigger on resize" (i.e. Pong), or
"I want to be able to see more things on resize" (i.e. KeyboardScroller).
See #65.

The `MouseSystem` and `CameraSystem` are changed accordingly.

Also created a separate `audio.go` file for Windows, to
be sure it also builds on Windows. Fixes #121.

### CODE REVIEW

**Comment by paked** [comment]
Yes. Demo would be great.

**Comment by matiwinnetou** [comment]
:+1:

**Comment by paked** [comment]
@EtienneBruines I'm happy to merge this in once you have time to clean up the merge conflicts.

I can make an example at a later date, just add an issue for it.

**Comment by EtienneBruines** [comment]
Updated to `master`, and changed some stuff. 

Please review again, and merge when satisfied. The (updated) first-post of this PR might help.  @paked @matiwinnetou

File: audio_windows.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	"log"
+
+	"github.com/paked/engi/ecs"
+)
+
+const (
+	defaultHeightModifier float32 = 1
+)
+
+// AudioComponent is a Component which is used by the AudioSystem
```
Comment (paked) [inline audio_windows.go]: This should be defined in a separate `audio.go`, also... Probably should make this file be ommitted from a non-windows build.

Comment (EtienneBruines) [inline audio_windows.go]: The `_windows` part of the filename already makes sure of that.

File: mouse.go [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+	Dragged bool
+	// Rightclicked is true whenever the space was rightclicked in this frame
+	Rightclicked bool
+	// Enter is true whenever the Mouse entered the space in that frame, but wasn't in that space on the previous frame
+	Enter bool
+	// Leave is true whenever the Mouse was in the space on the previous frame, but now isn't
+	Leave bool
+}
+
+func (*MouseComponent) Type() string {
```
Comment (paked) [inline mouse.go]: This needs a comment

File: mouse.go [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+// MouseSystem listens for mouse events, and changes value for MouseComponent accordingly
+type MouseSystem struct {
+	*ecs.System
+
+	mouseX    float32
+	mouseY    float32
+	mouseDown bool
+}
+
+func (*MouseSystem) Type() string {
```
Comment (paked) [inline mouse.go]: These functions need comments

File: mouse.go (hunk 2 above)
Comment (EtienneBruines) [inline mouse.go]: The `Type()` function? According to `golint`: yes, but tbh, if we were to actually comment *every* `Type()` function, it wouldn't make any sense..?

File: mouse.go [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+		return
+	}
+
+	// Reset some values
+	mc.Leave = false
+
+	mx := m.mouseX
+	my := m.mouseY
+
+	log.Println(mx, my, space.Position.X, space.Position.Y)
```
Comment (paked) [inline mouse.go]: Do we need this logging statement?

Comment (EtienneBruines) [inline mouse.go]: We most certainly do not; good catch, has to be removed.

**Comment by EtienneBruines** [comment]
Updated again.

File: audio_windows.go (hunk 1 above)
Comment (paked) [inline audio_windows.go]: Oh... Derp!

**Comment by paked** [comment]
LGTM.

**Pull request merged**